SConstruct

```diff
@@ -2888,6 +2888,23 @@ BOOST_AUTO_TEST_CASE(f)
 		if register_runtime_test_link_targets:
 			self.check_boost_test(context)
 
+	# dylibbundler is used to embed libraries into macOS app bundles,
+	# however it's only available on macOS.  While required for macOS
+	# builds, builds should not fail on other operating systems if
+	# it's absent.
+	@_custom_test
+	def _check_dylibbundler(self,context):
+		context.Display('%s: checking whether dylibbundler is installed...' % self.msgprefix)
+		if sys.platform != 'darwin':
```

Comment:
I think this test is incorrect.  A MacOS user can cross-compile for a non-MacOS target, and `sys.platform` would be `darwin` in that case because the Python interpreter is running on MacOS, but the desired code flow would be to skip the dylibbundler check, because dylibbundler is not required when cross-compiling for a non-MacOS target.

I suggest instead inspecting the target host_platform.

Reply:
You may find the design _guarded_test_windows decorator useful here.  Guarded tests are not even registered on platforms for which the guard is false, so there is no need for them to display a message.

Reply:
That makes sense, sure.

For the guarded tests, are those evaluated based on the target platform that's being compiled for, or the platform the compiler is running on?  In other words, if somebody is cross-compiling on a Mac for a non-macOS target, would a test that uses _guarded_test_darwin be skipped?

Reply:
That would depend on how you wrote the guard.  The existing `_guarded_test_windows` decorator inspects the `host_platform` which, following autotools convention, is the platform on which the built program is meant to run, rather than the platform on which SCons and the compiler are running.  Assuming you followed that pattern, then yes, a MacOS user cross-compiling to a non-MacOS target would skip the guarded test.

Definition of `_guarded_test_windows`:

https://github.com/dxx-rebirth/dxx-rebirth/blob/48589d0fb019eb971818c15665358e30fa778dc4/SConstruct#L416

Logic that uses the guards to exclude tests from execution:

https://github.com/dxx-rebirth/dxx-rebirth/blob/48589d0fb019eb971818c15665358e30fa778dc4/SConstruct#L606